src/AbstractMultilineWhitespaceBeforeChainedCallsEndCharacterFixer.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Graham Campbell <graham@alt-three.com>
+ * @author Egidijus Girčys <e.gircys@gmail.com>
+ *
+ * @internal
+ */
+abstract class AbstractMultilineWhitespaceBeforeChainedCallsEndCharacterFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface, WhitespacesAwareFixerInterface
+{
+    /**
+     * @internal
+     */
+    const STRATEGY_NO_MULTI_LINE = 'no_multi_line';
+
+    /**
+     * @internal
+     */
+    const STRATEGY_NEW_LINE_FOR_CHAINED_CALLS = 'new_line_for_chained_calls';
+
+    /**
+     * @var string
+     */
+    private $character;
```

Comment:
what is that character, semantically ?

Reply:
It's the character that ends a method calls chain, i.e. `;` or `,`. I'll rename the property to `$chainedCallsEndCharacter`.

Reply:
Property renamed.